test/dns_libvirt/dns_libvirt_provider_test.rb

```diff
@@ -0,0 +1,76 @@
+require 'test_helper'
+require 'dns_libvirt/dns_libvirt_plugin'
+require 'dns_libvirt/dns_libvirt_main'
+
+class DnsLibvirtProviderTest < Test::Unit::TestCase
+  def setup
+    fixture = <<XMLFIXTURE
+<network>
+  <name>default</name>
+  <domain name='local.lan'/>
+  <dns>
+    <host ip='192.168.122.1'>
+      <hostname>some.example.com</hostname>
+    </host>
+  </dns>
+  <ip address='192.168.122.0' netmask='255.255.255.0'>
+    <dhcp>
+      <range start='192.168.122.1' end='192.168.122.250'/>
+      <host mac='52:54:00:e2:62:08' name='some.example.com' ip='192.168.122.1'/>
+    </dhcp>
+  </ip>
+</network>
+XMLFIXTURE
+    @libvirt_network = mock()
+    @libvirt_network.stubs(:dump_xml).returns(fixture)
+    @subject = Proxy::Dns::Libvirt::Record.new(
+      :libvirt_network => @libvirt_network
+    )
+  end
+
+  def test_default_settings
+    ::Proxy::Dns::Libvirt::Plugin.load_test_settings({})
+    assert_equal 'default', Proxy::Dns::Libvirt::Plugin.settings.network
+  end
+
+  def test_provider_initialization
+    ::Proxy::Dns::Libvirt::Plugin.load_test_settings(:network => 'some_network')
+    assert_equal "some_network", Proxy::Dns::Libvirt::Record.new(:libvirt_network => @libvirt_network).network
```

Comment:
Ditto here, a test verifying that the correct network class was used to initialize the provider would be useful.